### PR TITLE
XWIKI-21878: Various close button modals do not use intended icons

### PR DESCRIPTION
**Pull request by Sereza7**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-21878
# Changes

## Description

* Removed the hard coded values from the templates.
* Made a REST API call from modal.js to access the current icontheme icon.

## Clarifications

* The only occurrence of `&times;` left in the project is the fallback value for the modal.js template.

# Screenshots & Video

All of those screenshots are taken after the changes in this PR. Hereafter are four of those. Among them, there's the "more difficult' cases with icons integrated in javascript (`modal.js` and `syntaxPicker.js`).
21878-moveAfterPR
21878-syntaxConvAfterPR
21878-previewDiffafterPR
21878-ckeditorModalAfterPR

# Executed Tests

Manual tests with FA4, see screenshots. I also checked the modal.js template behaved as expected with Silk, and it behaved as expected.
No test build conducted. From what I could see, this change does not reflect into any test (except the resources `xwiki-valid.html` and `xwiki-invalid.html`). Passes regular builds: `mvn clean install -f xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins; mvn clean install -f xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-webjar/` needed for testing on a live instance.

# Expected merging strategy

* Prefers squash: Yes 
* Backport on branches:
  * Only merge on master.

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js [hunk 1]
```diff
@@ -25,13 +25,40 @@ define('modalTranslationKeys', [], [
 
 define('modal', ['jquery', 'l10n!modal', 'bootstrap'], function($, translations) {
   'use strict';
-  var modalTemplate = 
+  // Fetch the cross icon from the icon theme to fill up the modal template.
+  let iconURL = `${XWiki.contextPath}/rest/wikis/${XWiki.currentWiki}/iconThemes/icons?name=cross`;
+  let iconRequest = new XMLHttpRequest();
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js:30]: any reason to not use a jquery `$.get` call here?

Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js:30]: No specific reason. I don't know much about async javascript and API calls yet. I'll spend some time to try and use jquery instead :+1:
Setting this PR as a draft until this is done

**Comment by Sereza7** [comment]
Used jquery `.get` instead of native javascript for the API call.

File: xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js [hunk 2]
```diff
@@ -25,33 +25,60 @@ define('modalTranslationKeys', [], [
...
-            '</button>' +
-            '<h4 class="modal-title"></h4>' +
-          '</div>' +
-          '<div class="modal-body"></div>' +
-          '<div class="modal-footer">' +
-            '<button type="button" class="btn btn-default" data-dismiss="modal">Cancel</button>' +
-            '<button type="button" class="btn btn-primary" disabled="disabled">OK</button>' +
-          '</div>' +
+  // Fetch the cross icon from the icon theme to fill up the modal template.
+  let iconURL = `${XWiki.contextPath}/rest/wikis/${XWiki.currentWiki}/iconThemes/icons?name=cross`;
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js:29]: I think the wiki name should be URL-encoded to protect against wiki names with special characters (not sure if we allow them, though).

File: xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js [hunk 3]
```diff
@@ -25,33 +25,60 @@ define('modalTranslationKeys', [], [
...
+        '<div class="modal-header">' +
+          '<button type="button" class="close" data-dismiss="modal" aria-label="Close">' +
+            closeButtonTemplate +
+          '</button>' +
+          '<h4 class="modal-title"></h4>' +
+        '</div>' +
+        '<div class="modal-body"></div>' +
+        '<div class="modal-footer">' +
+          '<button type="button" class="btn btn-default" data-dismiss="modal">Cancel</button>' +
+          '<button type="button" class="btn btn-primary" disabled="disabled">OK</button>' +
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js:67]: Do we have a Jira ticket to translate these buttons?

File: xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js (hunk 2 above)
Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-dialog/modal.js]: https://github.com/search?q=repo%3Axwiki%2Fxwiki-platform%20%24%7BXWiki.currentWiki%7D&type=code
It's already used a few times in the codebase.

I took the opportunity to fix those other uses in 4 files in https://github.com/xwiki/xwiki-platform/pull/2888/commits/217df8a8dc3c8c1f90f753607b729af16b6f7bdc .

I rebuilt the ckeditor plugin with the change, and it didn't break anything when testing on the main wiki :)

**Pull request merged**